drivers/stepper/h_bridge_stepper.c

```diff
@@ -268,6 +268,11 @@ static int h_bridge_stepper_run(const struct device *dev, const enum stepper_dir
 {
 	struct h_bridge_stepper_data *data = dev->data;
 
+	if (data->delay_in_ns == 0) {
```

Comment:
Could one also set an assert in the `set_microstep_interval` impl to check if the value is not set to 0?